TouchEvent/ViewController.swift

```diff
@@ -52,46 +57,61 @@ class ViewController: UIViewController, PHPickerViewControllerDelegate {
     }
     
     @IBAction func changeBackground() {
+        //PHPickerViewControllerを用意
         var configuration = PHPickerConfiguration()
+        
+        //選択できるアセットタイプを画像に限定
         let filter = PHPickerFilter.images
         configuration.filter = filter
         let picker = PHPickerViewController(configuration: configuration)
+        
+        //デリゲートを設定
         picker.delegate = self
+        
+        //フォトライブラリを呼び出す
         present(picker, animated: true)
     }
     
     @IBAction func save() {
+        //画面のスクリーンショットを撮影
         UIGraphicsBeginImageContextWithOptions(backgroundImageView.frame.size, false, 0.0)
         let context = UIGraphicsGetCurrentContext()!
         context.translateBy(x: -backgroundImageView.frame.origin.x, y: -backgroundImageView.frame.origin.y)
-        self.view.layer.render(in: context)
+        view.layer.render(in: context)
         let screenshot = UIGraphicsGetImageFromCurrentImageContext()
         UIGraphicsEndImageContext()
         
+        //フォトライブラリに保存
         UIImageWriteToSavedPhotosAlbum(screenshot!, nil, nil, nil)
     }
     
     @IBAction func undo() {
-        if !imageViewArray.isEmpty {
+        if imageViewArray.isEmpty {
+            //配列が空
```

Comment:
こんな感じでearly returnしてあげるのも良いかなと思うんだけど、おいしん的にどっちのほうが良さそう？
```suggestion
if imageViewArray.isEmpty { return }
```

Reply:
そうですね、分量的にもすぐ説明できるしearly returnにしちゃってこの機会に説明を載せるようにしたいと思います！